geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/DisconnectClientIntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+  private Socket socket;
+  private Cache cache;
+  private SecurityManager securityManager;
+
+  @Rule
+  public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+
+  @Before
+  public void setUp() throws Exception {
+    CacheFactory cacheFactory = new CacheFactory(new Properties());
+    cacheFactory.set(ConfigurationProperties.MCAST_PORT, "0");
+    cacheFactory.set(ConfigurationProperties.ENABLE_CLUSTER_CONFIGURATION, "false");
+    cacheFactory.set(ConfigurationProperties.USE_CLUSTER_CONFIGURATION, "false");
+
+    securityManager = mock(SecurityManager.class);
+    cacheFactory.setSecurityManager(securityManager);
+    when(securityManager.authenticate(any())).thenReturn(SECURITY_PRINCIPAL);
+    when(securityManager.authorize(eq(SECURITY_PRINCIPAL), any())).thenReturn(true);
+
+    cache = cacheFactory.create();
+
+    CacheServer cacheServer = cache.addCacheServer();
+    int cacheServerPort = AvailablePortHelper.getRandomAvailableTCPPort();
+    cacheServer.setPort(cacheServerPort);
+    cacheServer.start();
+
+    RegionFactory<Object, Object> regionFactory = cache.createRegionFactory();
+    regionFactory.setDataPolicy(DataPolicy.PARTITION);
+
+
+    System.setProperty("geode.feature-protobuf-protocol", "true");
+
+    socket = new Socket("localhost", cacheServerPort);
+
+    Awaitility.await().atMost(5, TimeUnit.SECONDS).until(socket::isConnected);
+
+    MessageUtil.performAndVerifyHandshake(socket);
+  }
+
+  @After
+  public void tearDown() {
+    cache.close();
+    try {
+      socket.close();
+    } catch (IOException ignore) {
+      // NOP
+    }
+  }
+
+  @Test
+  public void disconnectsFromServer() throws Exception {
+    authenticateWithServer();
+
+    final ClientProtocol.Message requestMessage = createRequestMessageBuilder(
+        ConnectionAPI.DisconnectClientRequest.newBuilder().setReason("Normal termination")).build();
+
+    final ClientProtocol.Message responseMessage = writeMessage(requestMessage);
+    assertEquals(responseMessage.toString(),
+        ClientProtocol.Message.MessageTypeCase.DISCONNECTCLIENTRESPONSE,
+        responseMessage.getMessageTypeCase());
+    final ConnectionAPI.DisconnectClientResponse disconnectClientResponse =
+        responseMessage.getDisconnectClientResponse();
+    assertNotNull(disconnectClientResponse);
+    assertEquals(true, disconnectClientResponse.getAcknowledged());
+  }
+
+  private void authenticateWithServer() throws IOException {
+    ClientProtocol.Message.Builder request = ClientProtocol.Message.newBuilder()
+        .setAuthenticationRequest(ConnectionAPI.AuthenticationRequest.newBuilder()
+            .putCredentials(ResourceConstants.USER_NAME, "someuser")
+            .putCredentials(ResourceConstants.PASSWORD, "somepassword"));
+
+    ClientProtocol.Message response = writeMessage(request.build());
+    assertEquals(response.toString(), true,
+        response.getAuthenticationResponse().getAuthenticated());
```

Comment:
AssertTrue.  Also seems like it would be easier to run this test without a security manager since that's not something covered by this test.

Reply:
All of the other integration tests for operation handlers use authentication so I'll leave that for consistency.